packages/core/src/trait_manager/model/Trait.ts

```diff
@@ -63,8 +64,15 @@ export default class Trait extends Model<TraitProperties> {
         case DataVariableType:
           this.dynamicVariable = new TraitDataVariable(this.attributes.value, { em: this.em, trait: this });
           break;
+        case ConditionalVariableType: {
+          const { condition, ifTrue, ifFalse } = this.attributes.value;
+          this.dynamicVariable = new DataCondition(condition, ifTrue, ifFalse, { em: this.em });
+          break;
+        }
         default:
-          throw new Error(`Invalid data variable type. Expected '${DataVariableType}', but found '${dataType}'.`);
+          throw new Error(
+            `Invalid data variable type. Expected '${DataVariableType} or ${ConditionalVariableType}', but found '${dataType}'.`,
+          );
```

Comment:
You know, I think we should avoid triggering errors here. The consumer could use any kind of custom value he wants (also objects).

Reply:
Done and I've also added a test for custom and `object` values.